src/main/webapp/placeGuides/MapPlaceGuideDisplayer.js

```diff
@@ -13,6 +13,42 @@ class MapPlaceGuideDisplayer {
     this.addNewPlaceGuidesToMap(placeGuides);
   }
 
+  remove(placeGuideId) {
+    this._placeGuidesOnMap[placeGuideId].remove();
+    delete this._placeGuidesOnMap[placeGuideId];
+  }
+
+  highlight(placeGuideId) {
+    this._placeGuidesOnMap[placeGuideId].highlight();
+  }
+
+  unhighlight(placeGuideId) {
+    this._placeGuidesOnMap[placeGuideId].unhighlight();
+  }
+
+  // This function sets the map bound to the minimum one
+  // which contains all the PlaceGuides.
+  adjustMapToShowAll() {
+    let mapBounds = new google.maps.LatLngBounds();
+    let guidesExist = false;
+    for (const placeGuideId in this._placeGuidesOnMap) {
+      if (this._placeGuidesOnMap.hasOwnProperty(placeGuideId)) {
```

Comment:
What are the cases where placeGuideId appears in the for...in iteration but not as hasOwnProperty? If there is a difference you could consider using Object.getOwnPropertyNames to simplify this code a bit and avoid the "let" variable.

Reply:
Done.